hyperspy/drawing/utils.py

```diff
@@ -131,7 +132,15 @@ def create_figure(window_title=None,
     fig : plt.figure
 
     """
-    fig = plt.figure(**kwargs)
+    if "ipympl" in mpl.get_backend() and preferences.Plot.enable_ipympl_plotting:
+        # this output is currently not used, just used for hiding.
+        # instead we get at the figure canvas widgets themselves later
+        output_widget = ipywidgets.widgets.Output()
+        with output_widget:
+            fig = plt.figure(**kwargs)
+    else:
+        output_widget = None
+        fig = plt.figure(**kwargs)
```

Comment:
So, the principal reason for using `with output_widget:` here is to prevent the figure from being displayed immediately. But the Output widget itself is a bit useless because we already have a good handle on the "figure widget" through the following two objects, which I get at in `mpl_he.py`:

`s._plot.navigator_plot.figure.canvas`
`s._plot.signal_plot.figure.canvas`

I'm looking at how difficult it would be to add the output widget as an argument. I agree it would be neater to not just throw it away, but I'm not sure how useful it is - unless, as you suggest it may be used for the histogram tool. (but I'm not really familiar with that problem)

Reply:
> I agree it would be neater to not just throw it away, but I'm not sure how useful it is

This would allow to put the figure in an specified output widget -> enable "custom plot layout" you mentioned above using hyperspy public API.